server/ec2spotmanager/cron.py

```diff
@@ -8,10 +12,16 @@
 from .CloudProvider.CloudProvider import INSTANCE_STATE, PROVIDERS, CloudProvider
 
 
+LOG = logging.getLogger("ec2spotmanager")
+
 STATS_DELTA_SECS = 60 * 15  # 30 minutes
 STATS_TOTAL_DETAILED = 24  # How many hours the detailed statistics should include
 STATS_TOTAL_ACCUMULATED = 30  # How many days should we keep accumulated statistics
 
+# How long check_instance_pools lock should remain valid. If the task takes longer than this to complete, the lock will
+# be invalidated and another task allowed to run.
+CHECK_POOL_LOCK_EXPIRY = 30 * 60
```

Comment:
In my testing it's taking <100ms. On our server now, each pool is taking around 1-200ms to update. I set this to 30 minutes as an unreasonably long timeout, in case something goes really wrong (like a worker gets SIGKILL) and the lock is not released. It should be released in all cases (normal and error) and this timeout should never be hit.

The file-based locks we were using with fasteners would release if the process went away for any reason because of posix semantics. We don't have that with Redis locks, so this is a safeguard.